src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -14,6 +14,23 @@ class AzureServiceClient < MsRest::ServiceClient
     # @return [String] api version of the Azure in string format.
     attr_accessor :api_version
 
+    # @return [Hash{String=>String}] default request headers for Azure requests
+    REQUEST_HEADERS = {
+        'x-ms-client-request-id' => SecureRandom.uuid,
```

Comment:
As I understand it, it's a unique number sent in the request. I reviewed this again and it does come from swagger, so I'll remove it from here, and just leave it up to the generator or the caller of the generic request to pass it, as necessary.